### PR TITLE
allow custimization of resource flags

### LINKED ISSUE #2746: Make controller threads configurable on the operator CR
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line:
>
> /kind bug

/kind enhancement

**What happened**:

Some users fine-tune the controller thread count. See #2648. Using manifests is already deprecated. We should not forget to move these options  to the operator install.

**What you expected to happen**:

That I can either set these specific options on the CR, or that we introduce a generic mapping between component commandline arguments and the CR, so that we don't have to map each one individually.

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- KubeVirt version (use `virtctl version`):
- Kubernetes version (use `kubectl version`):
- VM or VMI specifications:
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by ashleyschuett**
**What this PR does / why we need it**:

This PR allows users to define flags values for running the KubeVirt components in the KubeVirt CR. 
This is needed so that users that are deploying KubeVirt using the operator can set flags on resources at creation time. 

Update to PR https://github.com/kubevirt/kubevirt/pull/3402

**Which issue(s) this PR fixes**: 
Fixes #2746 

**Special notes for your reviewer**:

**Release note**:
```release-note
Set KubeVirt resources flags in the KubeVirt CR 
```

### CODE REVIEW

**Review by davidvossel — COMMENTED** [review]
I'm not opposed to making it easier for people to customize the component cli args. However I think we need to be careful how we document this feature.

So far our cli args haven't been something that are versioned as part of our API. By exposing them on the API, if we're not careful we're extending our API versioning to our cli args as well.

For example, if we decided to change the name of a CLI arg on virt-controller, and someone is using a KubeVIrt CR that set that arg's previous name, i'm pretty sure the controller will fail to boot entirely. It will complain about an unknown argument.

Here's my thought on how we can avoid this. I think this Flags map belongs in the CustomizeComponents struct next to Patches, and we should document that there are no guarantees around forward/backward compatibility if someone elects to use anything in the `CustomizeComponents` struct. 

sound reasonable?

File: staging/src/kubevirt.io/client-go/api/v1/types.go [hunk 1]
```diff
@@ -1453,9 +1453,20 @@ type KubeVirtSpec struct {
...
+
 	CustomizeComponents CustomizeComponents `json:"customizeComponents,omitempty"`
 }
 
+// +k8s:openapi-gen=true
+type Flags struct {
+	API        map[string]string `json:"api,omitempty"`
+	Controller map[string]string `json:"controller,omitempty"`
+	Handler    map[string]string `json:"handler,omitempty"`
+	Operator   map[string]string `json:"operator,omitempty"`
```
[the remark this answers is not in the document]
Comment (ashleyschuett) [reply staging/src/kubevirt.io/client-go/api/v1/types.go:1467]: You're right. This is my mistake. I for some reason thought we where already reconciling the operator deployment as well. But we don't. I'll remove this option.

**Comment by ashleyschuett** [comment]
@davidvossel thank you for the review! I have updated it with your suggestions

**Review by davidvossel — APPROVED** [review]
/lgtm
/approve

**Pull request merged**